src/algorithm/centroid.rs

```diff
@@ -0,0 +1,87 @@
+pub use types::{Coordinate, Point, LineString};
+
+pub trait Centroid {
+	/// Calculation the centroid, see: https://en.wikipedia.org/wiki/Centroid
+    fn centroid(&self) -> Result<Point, &'static str>;
+}
+
+impl Centroid for LineString {
+    /// Centroid on a LineString is the mean of the middle of the segment
+    /// weighted by the length of the segments:
+    /// +---------------------------------------+-+-+-+-+-+
+    ///                  centroid ^           ^ mean
+    ///
+    fn centroid(&self) -> Result<Point, &'static str> {
+        let vect = &self.0;
+        if vect.is_empty() {
+            return Err("Empty linestring");
+        }
+        if vect.len() == 1 {
+            return Ok(Point(Coordinate {
+                x: vect[0].lng(),
+                y: vect[0].lat(),
+            }));
+        } else {
+            let mut sum_x = 0.;
+            let mut sum_y = 0.;
+            let mut total_length = 0.;
+            for i in 0..vect.len() - 1 {
```

Comment:
I thought I remember an itertools stuff to do this, but I only found this:

https://github.com/slapresta/rust-iterslide

but It returns a list and not tuples :disappointed: